pandas/core/algorithms.py

```diff
@@ -1338,10 +1340,12 @@ class SelectNFrame(SelectN):
     nordered : DataFrame
     """
 
-    def __init__(self, obj, n: int, keep: str, columns):
+    def __init__(self, obj: DataFrame, n: int, keep: str, columns: IndexLabel):
         super().__init__(obj, n, keep)
         if not is_list_like(columns) or isinstance(columns, tuple):
             columns = [columns]
+
+        assert isinstance(columns, abc.Iterable)
```

Comment:
if adding an assert for mypy, append an inline comment `# for mypy` since we can't catch these when no longer necessary (i.e. if TypeGuard can narrow the types from the is_list_like condition.)

Alternatively, use a cast. we use cast elsewhere after `is_*` calls and we can detect when not needed with `warn_redundant_casts` mypy setting.

maybe Iterable from typing instead of collections but typing.Iterable is deprecated https://docs.python.org/3/library/typing.html#typing.Iterable

so okay to use typing.Iterable for consistency and also okay to keep the import from collections.

we will need to do a sweep at some point to migrate the deprecated uses.